.cursor/rules/guides/accessibility.mdc

```diff
@@ -0,0 +1,125 @@
+---
+description: React components accessibility principles
+globs: 
+alwaysApply: false
+---
+# Accessibility Guidelines
+
+## General Principles
+
+- All UI components must be accessible for users with disabilities
+- Follow WCAG 2.1 AA standards for accessibility
+- Always provide appropriate ARIA attributes for custom UI controls
+- Use semantic HTML elements whenever possible
+- Ensure keyboard navigation works for all interactive elements
+- Provide appropriate focus management
+
+## Component Guidelines
+
+### Using AccessibilityConfig
+
+Use the `accessibilityConfig` prop for components that support it:
+
+```tsx
+<Component
+  accessibilityConfig={{
+    ariaLabel: "Descriptive label",
+    ariaLabelledBy: "id-of-element-with-label",
+    ariaDescribedBy: "id-of-element-with-description",
+    role: "button", // or appropriate role
+  }}
+/>
+```
+
+### Buttons and Interactive Elements
+
+- All buttons need an accessible name via `aria-label`, `aria-labelledby`, or visible text content
```

Comment:
I would reformulate it, so visible text has highest priority and only if there is no visible text, aria attributes are used. It is imho error when you combine both visible text and aria prop.